src/glang/lib/std/map.c3

```diff
@@ -0,0 +1,102 @@
+@require_once "arithmetic.c3"
+@require_once "iterators.c3"
+@require_once "rb_tree.c3"
+@require_once "type_traits.c3"
+@require_once "wrappers.c3"
+
+typedef [K, V] MapNode : {
+    key   : K,
+    value : V
+}
+
+typedef [K, V] MapIter : RB_Iter<MapNode<K, V>>
+
+// FIXME <=> is untested and broken :(
+typedef [K, V] HasSpaceship<MapNode<K, V>> : HasSpaceship<K>
+
+@operator [K1, K2, V] == : (left : MapNode<K1, V>&, right : MapNode<K2, V>&) -> bool = {
+    return &left.key == &right.key;
+}
+
+@operator [K1, K2, V] != : (left : MapNode<K1, V>&, right : MapNode<K2, V>&) -> bool = {
+    return &left.key != &right.key;
+}
```

Comment:
Fortunately, the strange syntax stops being awkward when (if) we add automatic const ref promotion.